service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -179,23 +178,25 @@ private static boolean processHopByHopHeaders(HttpSession src, HttpSession dest)
         }
 
         // Add source session headers to destination session
-        for(Map.Entry<String, List<String>> e : src.getReadHeaders().entrySet()) {
+        for (Map.Entry<String, List<String>> e : src.getReadHeaders().entrySet()) {
             String name = e.getKey();
-            for(String value : e.getValue()) {
+            for (String value : e.getValue()) {
                 if (!hopByHopHeaders.contains(name)) {
-                    dest.addWriteHeader(name, value);
+                    if (name.equalsIgnoreCase(HttpHeaders.HEADER_HOST)) {
+                        dest.addWriteHeader(name, dest.getRemoteAddress().getResource().getAuthority());
+                    } else {
```

Comment:
@jitsni I have moved the fix to HttpConnectProcessor, it seemed to be a better place for it. And also modifies only the HttpRequestMessage with the updated header and not the headers from the session. Let me know what you think.